### ISSUE TITLE
Update without payment doesn't work correctly

### ISSUE THREAD

**Issue by ianleeder** [eWAYPayment/eway-rapid-net#11]
Me again.

As per #9, I want to run a charge against the new payment details to ensure (at update time) there won't be any issues when next due for a recurring payment.  And as suggested, I'm doing an update _without_ payment details, followed by a recurring payment.

I've updated the same [Gist](https://gist.github.com/ianleeder/2bf770ff70a5a9288f13) as used in the previous issue (only new features added, no existing code changed).  And it partially works.

If you register with a credit card, you can use the new method (`Update details without payment THEN charge`) to change credit card details.  This can be verified in MYeWAY that the charge went to the new card, and that the new card is stored against the token.

If you register with a credit card and use the new method to update to Paypal, this also works.  From the MYeWAY home page we can see that the most recent transaction was against a Paypal account.  We _can't_ see the Paypal information against the token (see eWAY Support Case #00373651), but that's nothing new.

If you register with Paypal and use the new method to update to a different Paypal account, this works.

**_However**_, if you register with Paypal and use the new method to update to _card_, it **does not work**.  The MYeWAY home page shows the charge still went to Paypal.  As above we are unable to see what is stored on the token, however running another recurring charge still goes to Paypal.

TLDR: You can't update stored Paypal information to Credit card.
CC -> CC ok
CC -> PP ok
PP -> PP ok
PP -> CC fails.

My first thought was a timing issue: the charge came through before the update was finalised.  However I wouldn't expect it to only occur for that one combination.  I would expect that it would either be all cases _from_ PP, or all cases _to_ CC.  Also, the next recurring charge against that token (seconds later) still goes against Paypal.

**Comment by ianleeder** [comment]
This is so much worse than I thought.  It has nothing to do with submitting the payment.

If you have Paypal on record (either registered or updated to Paypal), and you update details using a credit card it will show the new card in MYeWAY, and on customer token queries, **but recurring payments will still be charged to Paypal**.

This is regardless of whether the update was with or without payment.  If it was with payment, _that_ payment is charged to the card (this is why I missed it earlier), but subsequent recurring payments go to Paypal.

This has gone from being an inconvenience to being a showstopper.  At this point, it looks like I have to cut support for Paypal, which is devastating since this was one of the reasons we went with eWAY.

I seriously hope this is only an SDK issue (as that would be an easier fix), or a sandbox issue.  Unfortunately I suspect the problem is in the back-end servers, since they don't deal with Paypal well.

**Comment by ianleeder** [comment]
Finally managed to test this on the live servers (needed to enable Reference Transactions on our Paypal account).  Unfortunately the same thing happens on a live server.

Which leaves my only "out" as this being an SDK error.  However since the customer payment details update correctly in MYeWAY I can't see any way this could be.

**Comment by incarnate** [comment]
Thanks for the detailed reports again Ian. I've also tested using curl and found the same, so definitely not an SDK issue as you noted.

I see you've submitted a case about the issue and I'll follow up on that to ensure our developer team are aware of it.

In the meantime, the only workaround I can think of would be to avoid using the update function for now. If a customer does want to update their details, make it an explicit choice in the process and generate a new token rather than updating the old.

**Comment by ianleeder** [comment]
Yep, that's exactly what we came up with too.  Every update = new token.  Hence I logged _another_ case (#00379331) about disabling tokens programmatically.

Thanks for verifying btw.  I couldn't see how but I always have a niggling feeling that it's just me doing something stupid and I'll end up with egg on my face.

I'll close the issue here since it's not an SDK issue.

**Comment by incarnate** [comment]
Thanks Ian - you may be interested in voting on and following [this suggestion](http://ideas.myeway.com/forums/181167-developer-forum-eway/suggestions/4959721-add-deletecustomer-method-to-rapid-3-0-token-pay) for disabling token customers via the API in our ideas forum.
